tools/githelper.py

```diff
@@ -40,9 +40,15 @@ def get_changed_lines(
     # here. Also, if we run 2 hooks that modify the files, the second hook
     # should work on the diff that includes the unstaged changes made by the
     # first hook, not the original diff.
-    cmd = ["git", "diff", "--patch", "--unified=0", changeset]
+    cmd = ["git", "diff", "--patch", "--unified=0", "--no-relative", changeset]
     if include_files:
-        cmd.extend(["--", *include_files])
+        toplevel_path = get_toplevel_path()
+        cmd.extend(
+            [
+                "--",
+                *(fn for fn in include_files if fn.startswith(toplevel_path)),
```

Comment:
When does git diff return files  that are not in the repository? 
A comment would be helpful.

Reply:
I removed that for now, because it requires a newer git version. I had this when using `git worktree`, when it returned a file in the `.git` dir of the original repo (not the worktree)